addons/edi/models/edi_gateway.py

```diff
@@ -216,6 +221,28 @@ def _compute_ssh_host_fingerprint(self):
             else:
                 gw.ssh_host_fingerprint = None
 
+
+    @api.depends("outgoing_server")
+    def compute_outgoing_server_id(self):
+        """Get the outgoing server from the smtp host name"""
+        MailServer = self.env["ir.mail_server"]
+        if self.outgoing_server:
+            server = MailServer.search([("smtp_host", "=", self.outgoing_server)])
+            if server:
+                # If there is more than one server with the same address choose the lowest id
+                if len(server) > 1:
```

Comment:
This if is not needed.

Reply:
The if is needed if someone makes multiple outgoing servers with the same host as it returns both.
Changed it to limit 1, and order by in the search. People shouldn't be making multiple records of the same outgoing server (unless I guess they want different config but not sure why)

Reply:
My bad, I wasn't clear, I meant we could have used `server[0] ` directly, but limit=1 is better :)